smart-py/marketplace.py

```diff
@@ -1,65 +1,954 @@
-        
+import smartpy as sp
+
 class Marketplace(sp.Contract):
     def __init__(self, objkt, metadata, manager, fee):
         self.init(
             objkt = objkt,
             metadata = metadata,
             manager = manager,
-            swaps = sp.big_map(tkey=sp.TNat, tvalue=sp.TRecord(issuer=sp.TAddress, objkt_amount=sp.TNat, objkt_id=sp.TNat, xtz_per_objkt=sp.TMutez, royalties=sp.TNat, creator=sp.TAddress)),
+            swaps = sp.big_map(
+                tkey=sp.TNat,
+                tvalue=sp.TRecord(
+                    issuer=sp.TAddress,
+                    objkt_amount=sp.TNat,
+                    objkt_id=sp.TNat,
+                    xtz_per_objkt=sp.TMutez,
+                    royalties=sp.TNat,
+                    creator=sp.TAddress
+                )
+            ),
             counter = 500000,
             fee = fee
-            )
-            
+        )
+
     @sp.entry_point
     def swap(self, params):
-        sp.verify((params.objkt_amount > 0) & ((params.royalties >= 0) | (params.royalties <= 250)))
-        self.fa2_transfer(self.data.objkt, sp.sender, sp.self_address, params.objkt_id, params.objkt_amount)
-        self.data.swaps[self.data.counter] = sp.record(issuer=sp.sender, objkt_amount=params.objkt_amount, objkt_id=params.objkt_id, xtz_per_objkt=params.xtz_per_objkt, royalties=params.royalties, creator=params.creator)
+        sp.verify(
+            (params.objkt_amount > 0) &
+            (
+                (params.royalties >= 0) & (params.royalties <= 250)
+            )
+        )
+
+        # TODO
+        # i want to validate the royalties here to make sure they match
+        # what is set on the objkt when it is minted
+
+        self.fa2_transfer(
+            self.data.objkt,
+            sp.sender,
+            sp.self_address,
+            params.objkt_id,
+            params.objkt_amount
+        )
+
+        self.data.swaps[self.data.counter] = sp.record(
+            issuer=sp.sender,
+            objkt_amount=params.objkt_amount,
+            objkt_id=params.objkt_id,
+            xtz_per_objkt=params.xtz_per_objkt,
+            royalties=params.royalties,
+            creator=params.creator
+        )
+
         self.data.counter += 1
-    
+
     @sp.entry_point
     def collect(self, params):
         sp.verify(
             # verifies if tez amount is equal to price per objkt
```

Comment:
Adjust comment that there is also a check to ensure there are more than 0 objkt's to collect.

Reply:
updated